File: w09/d01/joshua_oynick/mail_items/spec/package_spec.rb
require_relative 'spec_helper'
require_relative '../lib/postcard'
require_relative '../lib/mailitem'
require_relative '../lib/package'

describe Package do

  before(:each) do
    @package = Package.new
  end

  it "inherits MailItem" do
    expect(Package.superclass).to eq(Mailitem)
  end

  it "has a content array" do
    # package = Package.new
    expect(@package.content.is_a?(Array)).to eq(true)
  end

  it 'returns true if fragile' do
    expect(@package.is_fragile).to eq false
  end

  it 'returns true if it is broken' do
    expect(@package.is_broken).to eq false
  end

  it 'responds to shaking' do
    expect(@package).to respond_to :shake
  end

  it 'will be broken if it is fragile and is shaken' do
    @package.is_fragile = true
    @package.shake
    expect(@package.is_broken).to eq true
  end

  it 'responds to is_broken?' do
    expect(@package).to respond_to(:is_broken?)
  end




end

File: w09/d03/james_rothpearl/mail_objects_app/spec/javascripts/package_spec.js
describe("package", function(){

  var pckg;

  beforeEach(function(){
    pckg = new Package();
  });

  it("has a content array", function(){
    expect(pckg).toBeArray;
  });

  it("has a fragile boolean", function(){
    expect(pckg).toBeBoolean;
  });

  it("has a broken boolean", function(){
    expect(pckg).toBeBoolean;
  });

  it("has a shake method that sets fragile and broken to true", function(){
    pckg.fragile = true;
    pckg.shake();
    expect(pckg.broken).toBe(true)
  })

  it("has an isBroken method that returns whether it is broken or not", function(){
    pckg.fragile = true;
    pckg.shake();
    expect(pckg.isBroken()).toBe(true)
  })


})
